src/peft/peft_model.py

```diff
@@ -650,6 +677,84 @@ def _split_kwargs(cls, kwargs: Dict[str, Any]):
 
         return hf_hub_download_kwargs, other_kwargs
 
+    def _update_offload(self, offload_index: Dict[Dict[str:str]], adapters_weights: Dict[str : torch.tensor]):
+        """
+        Update the offload_index and safetensors files for loading and mergine PeftModels with disk-offloaded modules.
+
+        Args:
+            offload_index (Dict[str: str]):
+                Dictionary of disk-offloaded modules with their metadata and safetensors filenames
+            adapters_weights (Dict[str: torch.tensor]):
+                Dictionary of Peft adapter module names and weights
+        """
+
+        prefix = "base_model.model."
+        # rename offload index weight and model names
+        adapter_names = list(self.peft_config.keys())
+        for adapter_name in adapter_names:
+            target_modules = self.peft_config[adapter_name].target_modules
+            if offload_index:
+                keys = list(offload_index.keys())
+                block_id = keys[0].split(".")[0] + "."  # for writing safetensors key
+
+                # replace original offload index keys with PeftModel keys
+                for key in keys:
+                    if any(module in key for module in target_modules):
+                        suffix_pos = key.rfind(".")
+                        new_key = prefix + key[:suffix_pos] + ".base_layer" + key[suffix_pos:]
+                    else:
+                        new_key = prefix + key
+                    offload_index[key]["weight_name"] = new_key
+                    offload_index[new_key] = offload_index[key]
+                    del offload_index[key]
+
+                files_seen = set()
+                # rename safetensors for dispatch
+                for new_key in list(offload_index.keys()):
+                    fname = offload_index[new_key]["safetensors_file"]
+
+                    # make a new file name
+                    new_fname_list = list(fname.split(os.sep))
+                    for i, name in enumerate(new_fname_list):
+                        if "--" in name:
+                            new_fname_list[i] += "-peft"
```

Comment:
We can `break` after the first match, right?

Reply:
Good idea and yes indeed, only the `models--xxxxx` dirname is changed